pep-0710.rst

```diff
@@ -0,0 +1,416 @@
+As in the :ref:`direct URL origin specification<packaging:direct-url>`, the ``url`` value
+MUST be stripped of any sensitive authentication information for security reasons.
+
+The user:password section of the URL MAY however be composed of environment
+variables, matching the following regular expression:
+
+.. code-block:: text
+
+    \$\{[A-Za-z0-9-_]+\}(:\$\{[A-Za-z0-9-_]+\})?
+
+Additionally, the user:password section of the URL MAY be a well-known,
+non-security sensitive string. A typical example is ``git`` in the case of an
+URL such as ``ssh://git@gitlab.com``.
+
+The value of ``archive_info`` MUST be a dictionary with a single key
+``hashes``.  The value of ``hashes`` is a dictionary mapping hash function names to a
+hex-encoded digest of the file referenced by the ``url`` value. Multiple hashes
+can be included, and it is up to the consumer to decide what to do with
+multiple hashes (it may validate all of them or a subset of them, or nothing at
+all).
+
+Each hash MUST be one of the single argument hashes provided by
+:data:`py3.11:hashlib.algorithms_guaranteed`, excluding ``sha1`` and ``md5`` which MUST NOT be used.
+As of Python 3.11, with ``shake_128`` and ``shake_256`` excluded
+for being multi-argument, the allowed set of hashes is:
+
+.. code-block:: python
+
+  >>> import hashlib
+  >>> sorted(hashlib.algorithms_guaranteed - {"shake_128", "shake_256", "sha1", "md5"})
+  ['blake2b', 'blake2s', 'sha224', 'sha256', 'sha384', 'sha3_224', 'sha3_256', 'sha3_384', 'sha3_512', 'sha512']
+
+Each hash MUST be referenced by the canonical name of the hash, always lower case.
+
+Hashes ``sha1`` and ``md5`` MUST NOT be present, due to the security
+limitations of these hash algorithms. Conversely, hash ``sha256`` SHOULD
+be included.
+
+Installers that cache distribution packages from an index SHOULD keep
+information related to the cached distribution artifact, so that
+the ``provenance_url.json`` file can be created even when installing distribution packages
+from the installer's cache.
+
+Backwards Compatibility
+=======================
+
+Since this PEP specifies a new file in the ``*.dist-info`` directory, there are
+no backwards compatibility implications to consider in the ``provenance_url.json``
+file itself. Also, this proposal does not make any changes to the
+``direct_url.json`` described in :pep:`610` and
+:ref:`its corresponding canonical PyPA spec <direct-url>`.
+
+The content of ``provenance_url.json`` file was designed in a way to eventually
+allow installers reuse some of the logic supporting ``direct_url.json`` when a
+direct URL refers to a source archive or a wheel.
+
+Security Implications
+=====================
+
+One of the main security features of the ``provenance_url.json`` file is the
+ability to audit installed artifacts in Python environments. Tools can check
+which Python package indexes were used to install Python :term:`Distribution Packages`
```

Comment:
```suggestion
which Python package indexes were used to install Python distribution packages
```

Oops, forgot about the plural...simpler to just de-link it here, I guess.

Reply:
This seems to work:

```suggestion
which Python package indexes were used to install Python :term:`distribution packages <Distribution Package>`
```

From rst docs:

> You can link to a term in the glossary while showing different text in the topic by including the term in angle brackets.

Thanks for all your suggestions!

Reply:
Yeah, I know; I was just too lazy to type it out twice :)